pkg/image/oci/registry_provider.go

```diff
@@ -0,0 +1,76 @@
+package oci
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	"github.com/anchore/stereoscope/internal/log"
+	"github.com/anchore/stereoscope/pkg/file"
+	"github.com/anchore/stereoscope/pkg/image"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+// RegistryImageProvider is a image.Provider capable of fetching and representing a container image fetched from a remote registry (described by the OCI distribution spec).
+type RegistryImageProvider struct {
+	imageStr        string
+	tmpDirGen       *file.TempDirGenerator
+	registryOptions *image.RegistryOptions
+}
+
+// NewRegistryImageProvider creates a new provider instance for a specific image that will later be cached to the given directory.
+func NewRegistryImageProvider(imgStr string, tmpDirGen *file.TempDirGenerator, registryOptions *image.RegistryOptions) *RegistryImageProvider {
+	return &RegistryImageProvider{
+		imageStr:        imgStr,
+		tmpDirGen:       tmpDirGen,
+		registryOptions: registryOptions,
+	}
+}
+
+// Provide an image object that represents the cached docker image tar fetched a registry.
+func (p *RegistryImageProvider) Provide() (*image.Image, error) {
+	log.Debugf("pulling image info directly from registry image=%q", p.imageStr)
+
+	imageTempDir, err := p.tmpDirGen.NewTempDir()
+	if err != nil {
+		return nil, err
+	}
+
+	ref, err := name.ParseReference(p.imageStr)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse registry reference=%q: %+v", p.imageStr, err)
+	}
+
+	img, err := remote.Image(ref, prepareRemoteOptions(ref, p.registryOptions)...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create image from registry: %+v", err)
+	}
+
+	return image.NewImage(img, imageTempDir), nil
+}
+
+func prepareRemoteOptions(ref name.Reference, registryOptions *image.RegistryOptions) []remote.Option {
+	var opts []remote.Option
+	if registryOptions.InsecureSkipTLSVerify {
```

Comment:
nit: if the configuration of `tls.Config` is `InsecureSkipVerify` it would be easier to reason about this if it is named the same

Reply:
Without `TLS` in the name it would be hard to reason about what verification is being skipped. Take for example the stdlib `tls.Config`:
```golang
tls.Config{InsecureSkipVerify: true}
```
in this example it's relatively clear that the verification being skipped somehow relates to TLS, whereas for the registry options it would not be:
```golang
image.RegistryOptions{InsecureSkipVerify: true}
```
this could be misconstrued as skipping some sort of verification relating to the image or the registry, but it's difficult to pin down that it's about TLS here. 

I changed the config option to be more in line with how'd you'd configure `kubectl` which takes a `--insecure-skip-tls-verify` option for similar behavior.